tests/suites/os/tests/secureboot/index.js

```diff
@@ -1,145 +1,339 @@
 'use strict';
 
-module.exports = {
-	title: 'secure boot tests',
-	tests: [
-		{
-			title: 'check secure boot',
-			run: async function(test) {
-				const securebootEfiVarPath = '/sys/firmware/efi/efivars/SecureBoot-*';
+const fs = require('fs');
+const fse = require('fs-extra');
+const securebootEfiVarPath = '/sys/firmware/efi/efivars/SecureBoot-*';
 
-				const securebootSupported = await this.worker.executeCommandInHostOS(
-					['ls', securebootEfiVarPath, '&>/dev/null',
-						'&&', 'echo', 'true',
-						'||', 'echo', 'false'
-					],
-					this.link,
-				).then(output => { return output === 'true'});
-
-				// https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Checking_Secure_Boot_status
-				const securebootEnabled = await this.worker.executeCommandInHostOS(
-						[
-							'od', '--address-radix=n', '--format=u1', securebootEfiVarPath,
-								'|', 'tr', '-s', "' '",
-								'|', 'cut', "-d' '", '-f6'
-						],
-						this.link,
-				).then(output => output === '1' );
-
-				// Used after tampering with verified boot files
-				const resetWorker = async () => {
-					await this.worker.off();
-					await this.worker.flash(this.os.image.path);
-					await this.worker.on();
-
-					await this.worker.addSSHKey(this.sshKeyPath);
-				}
+class secureBoot {
+	constructor(test, worker, suite, imagePath, module = {"name": "", "headersVersion": ""}) {
+		this.test = test;
+		this.worker = worker;
+		this.link = suite.context.get().link;
+		this.imagePath = imagePath;
+		this.sshKeyPath = suite.context.get().sshKeyPath;
+		this.utils = suite.context.get().utils;
+		this.module = module;
+		this.tmpDir = suite.options.tmpdir;
+	}
 
-				const waitForSerialOutput = async(pattern, slice=0, retries=60, delay=1000) => this.utils.waitUntil(
-					async () => this.worker.fetchSerial().then(
-						serialLogs => serialLogs.split('\n')
-																		.slice(slice)
-																		.join('\n')
-																		.match(pattern)
-					), false, retries, delay
-				);
+	async resetDUT() {
+		await this.worker.off();
+		await this.worker.flash(this.imagePath);
+		await this.worker.on();
 
-				const testFullDiskEncryption = async () => Promise.all(
-					[
-						{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
-						{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
-						{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
-						{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
-					].map(args => test.resolveMatch(
-							this.worker.executeCommandInHostOS(
-								[
-									'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
-									'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
-								],
-								this.link,
-							),
-							/crypt/,
-							`${args.name} is encrypted`,
-						)
-					)
-				);
+		await this.worker.addSSHKey(this.sshKeyPath);
```

Comment:
I know this was in the old implementation , but I'm struggling to understand why its here - this sends the SSH key to the worker so you can ssh from worker -> DUT . As far as I can see we don't change any ssh keys here or remove any, so it feels like its probably not needed

Reply:
You're probably right, I'll try to remove it.